src/main/java/io/pinecone/clients/Inference.java

```diff
@@ -0,0 +1,94 @@
+package io.pinecone.clients;
+
+import org.openapitools.control.client.ApiClient;
+import org.openapitools.control.client.ApiException;
+import org.openapitools.control.client.api.InferenceApi;
+import org.openapitools.control.client.model.EmbedRequest;
+import org.openapitools.control.client.model.EmbedRequestInputsInner;
+import org.openapitools.control.client.model.EmbedRequestParameters;
+import org.openapitools.control.client.model.EmbeddingsList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * The Inference class provides methods to interact with Pinecone's embedding API through the Java SDK. It allows users
```

Comment:
I'd also consider linking out to documentation in this docstring, esp for the parameters, since it's unclear if those are just example params or mandatory params etc